torch/csrc/jit/script/ir_emitter.cpp

```diff
@@ -2735,6 +2738,104 @@ struct to_ir {
+    if (apply.inputs().size() < 2 || apply.inputs().size() > 4) {
+      throw ErrorReport(apply)
+          << "Possible forms of call to rpc_async(..) are\n"
+          << "rpc_async(dst_worker_name, user_callable, args, kwargs)\n"
+          << "rpc_async(dst_worker_name, user_callable, args)\n"
+          << "rpc_async(dst_worker_name, user_callable)\n"
+          << "Now the number of arguments is " << apply.inputs().size();
+    }
+    if (apply.attributes().size() != 0) {
+      throw ErrorReport(apply)
+          << "rpc_async(dst_worker_name, user_callable, args, kwargs)"
+          << "does not support kwargs yet";
+    }
+    // TODO: Make rpc_async(..) support taking kwargs,
+    // like rpc_async(to=, dst_worker_name="", args=(), kwargs={})
+
+    auto& input_trees = apply.inputs().tree()->trees();
+    Value* dst_worker_name_value = emitExpr(Expr(input_trees[0]));
+    std::shared_ptr<FunctionValue> user_callable_sugared_value =
+        std::dynamic_pointer_cast<FunctionValue>(
+            emitSugaredExpr(Expr(input_trees[1]), 1));
+    // If `kwargs` is an empty dict, users are allowed to not pass `kwargs`.
+    // If `args` and `kwargs` are an empty tuple and an empty dict,
+    // respectively, users are allowed to not pass `args` and `kwargs`.
+    TreeList args_kwargs_trees(input_trees.begin() + 2, input_trees.end());
+
+    // Get user function.
+    auto& callablePtrs = user_callable_sugared_value->callees();
+    TORCH_INTERNAL_ASSERT(
+        callablePtrs.size() == 1,
+        "User-provided callable size should be 1. Now it's",
+        callablePtrs.size())
+    Function* callablePtr = callablePtrs.at(0);
+
+    const SourceRange& loc = apply.range();
+    auto graphPtr = method.graph();
+
+    // Graph insert the QualifiedName as an constant input IR Value.
+    const auto& qualname = callablePtr->qualname();
+    IValue userCallableQualNameIValue(qualname.qualifiedName());
+    Value* userCallableQualNameValue =
+        graphPtr->insertConstant(userCallableQualNameIValue, loc);
+
+    // Graph insert a Node, prim::rpc_async jit::Operator, to the Graph.
+    Node* rpc_async_node =
+        graphPtr->insertNode(graphPtr->create(prim::rpc_async, 1))
+            ->setSourceRange(loc);
+    {
+      WithInsertPoint insert(rpc_async_node);
+      rpc_async_node->addInput(dst_worker_name_value);
+      rpc_async_node->addInput(userCallableQualNameValue);
+
+      for (const auto& tree : args_kwargs_trees) {
+        rpc_async_node->addInput(emitExpr(Expr(tree)));
+      }
+    }
+    Value* rpc_async_node_output = rpc_async_node->output();
+
+    // Set output type from FunctionSchema.
+    const auto& functionSchema = callablePtr->getSchema();
+    const std::vector<Argument>& returns = functionSchema.returns();
+    TORCH_INTERNAL_ASSERT(returns.size() == 1);
+    auto output_type = returns[0].type();
+    rpc_async_node_output->setType(FutureType::create(output_type));
+
+    // Match FunctionSchema.
```

Comment:
Let's first do the check of `Match FunctionSchema`, then if it's valid, we insert the prim node into the graph.

Reply:
Updating.